### ISSUE TITLE
Cleanup prometheus metrics

### ISSUE THREAD

**Issue by quadriq** [coturn/coturn#1235]
Hallo,
I configured prometheus metrics and I can see them by accessing this url:
http://localhost:9641/metrics

however I noticed, that displayed metrics are growing with the time, and getting deleted only after turnserver restart.
Otherwise the output getting bigger and bigger. Which make the tool that collects those metrics (I use telegraf) slower and slower.

Is there an option to specify how long metrics getting displayed in the /metrics endpoint? Maybe a kind of "retention policy" parameter?

**Comment by eakraly** [comment]
Hi @quadriq 

No, there is no such thing (in coturn at least) as "cleanup" of old metrics - they are reported until restart.
What makes the output huge is the use of `--prometheus-username-labels` flag (it also looks like a memory leak).

Since `--prometheus-username-labels` is off by default - how is that you have your output ever growing? Do you have it on? Or maybe you are using an older version of coturn which did not have the flag?

**Comment by quadriq** [comment]
yes, I am using `prometheus-username-labels` in my config. Otherwise how could I determinate how many traffic each user has used?

**Comment by eakraly** [comment]
@quadriq Great question

Prometheus is a tool for service monitoring, not business metrics or billing.
For that you should use something else.
Take a look at redis interface - it may be more helpful in this case

**Comment by quadriq** [comment]
I was able to connect redis. But it does not display any info about the traffic.
e.g. the only info available on redis:

```
 get "turn/realm/myrealm/user/myuser/allocation/003000000000000002/status"
"new lifetime=600, type=UDP, local=207.244.226.110:3478, remote=91.5.3.238:50748, ssl=NONE, cipher=NONE"
```

**Comment by eakraly** [comment]
@quadriq try this one https://github.com/coturn/coturn/blob/master/turndb/schema.stats.redis#L18C45-L18C53

**Comment by eakraly** [comment]
Not an issue - as designed
